core/shared/src/main/scala-2.13+/fs2/Chunk213And3Compat.scala

```diff
@@ -120,7 +120,7 @@ private[fs2] final class ChunkAsSeq[+O](
     chunk.toVector
 
   override def toString: String =
-    chunk.toString
+    chunk.iterator.mkString("ChunkAsSeq(", ", ", ")")
```

Comment:
I don't think the toString should leak the implementation class. For example:

```scala
scala> val c = Chunk(1, 2, 3)
val c: fs2.Chunk[Int] = Chunk(1, 2, 3)

scala> val c2 = c ++ c
val c2: fs2.Chunk[Int] = Chunk(1, 2, 3, 1, 2, 3)

scala> val first4 = c2.take(4)
val first4: fs2.Chunk[Int] = Chunk(1, 2, 3, 1)

scala> (c.getClass, c2.getClass, first4.getClass)
val res3: (Class[? <: fs2.Chunk[Int]], Class[? <: fs2.Chunk[Int]],
  Class[? <: fs2.Chunk[Int]]
) = (class fs2.Chunk$ArraySlice,class fs2.Chunk$Queue,class fs2.Chunk$Queue)
```

Reply:
@mpilquist Well then what should we use then?
The only two other options are `Seq` and `Chunk`, and I don't think either is a good option.
+ `Chunk`: Since `Chunk` is not a `Seq`, then doing `aSeq.toString` printing `Chunk(...)` feels misleading.
+ `Seq`: AFAIK _(I could be wrong in this)_ there is no a single type in the stdlib printing `Seq(...)` 

Also, I just realized that if we don't override then we would have got the same result: https://github.com/scala/scala/blob/v2.13.11/src/library/scala/collection/Iterable.scala#L78

Reply:
Disregard, as a toString for the `Seq` subtype, this is fine. I was thinking it was a new chunk subtype when I commented.

Reply:
@mpilquist I just noticed that `Chunk.empty` breaks this rule, and rather prints `empty` instead of `Chunk()`
Is that intentional?

Reply:
@mpilquist would you like me to remove that as well on this PR?
Or are we afraid that would break some user code?

Reply:
Sure